lib/modules/manager/unity3d/extract.ts

```diff
@@ -0,0 +1,46 @@
+import { regEx } from '../../../util/regex';
+import { Unity3dDatasource } from '../../datasource/unity3d';
+import type { PackageDependency, PackageFileContent } from '../types';
+
+const supportedKeys = ['m_EditorVersionWithRevision', 'm_EditorVersion'];
+
+const UnityVersionRegex = regEx(/^(?<depName>.+): (?<currentValue>.+)/);
+
+function extractVersions(content: string): PackageDependency[] {
+  const deps: PackageDependency[] = [];
+
+  for (const line of content.split('\n')) {
+    const matches = UnityVersionRegex.exec(line);
+
+    if (!matches?.groups) {
+      continue;
+    }
+
+    const key = matches.groups.depName;
+    const value = matches.groups.currentValue;
+
+    if (!supportedKeys.includes(key)) {
+      continue;
+    }
+
+    deps.push({
+      currentValue: value,
+      datasource: Unity3dDatasource.id,
+      depName: key,
```

Comment:
This will put both version changes in the same branch and same PR, plus the PR title will show `Unity3D Editor`
```suggestion
     depName: 'Unity3D Editor',
      packageName: key,
```